system-tests/tests/smoke/cre/v2_vault_don_test.go

```diff
@@ -154,30 +173,36 @@ func fetchVaultPublicKey(t *testing.T, gatewayURL string) (publicKey string) {
 	return publicKeyResponse.PublicKey
 }
 
-func executeVaultSecretsCreateTest(t *testing.T, encryptedSecret, secretID, owner, gatewayURL string) {
+func executeVaultSecretsCreateTest(t *testing.T, encryptedSecret, secretID, owner, gatewayURL string, opts *bind.TransactOpts, wfRegistryContract *workflow_registry_v2_wrapper.WorkflowRegistry) {
 	framework.L.Info().Msg("Creating secret...")
 
 	uniqueRequestID := uuid.New().String()
 
-	secretsCreateRequest := jsonrpc.Request[vaultcommon.CreateSecretsRequest]{
+	secretsCreateRequest := vaultcommon.CreateSecretsRequest{
+		RequestId: uniqueRequestID,
+		EncryptedSecrets: []*vaultcommon.EncryptedSecret{
+			{
+				Id: &vaultcommon.SecretIdentifier{
+					Key:   secretID,
+					Owner: owner,
+					// Namespace: "main", // Uncomment if you want to use namespaces
+				}, // Note: Namespace is not used in this test, but can be added if needed
+				EncryptedValue: encryptedSecret,
+			},
+		},
+	}
+	secretsCreateRequestBody, err := json.Marshal(secretsCreateRequest)
+	require.NoError(t, err, "failed to marshal secrets request")
+	secretsCreateRequestBodyJson := json.RawMessage(secretsCreateRequestBody)
+	jsonRequest := jsonrpc.Request[json.RawMessage]{
 		Version: jsonrpc.JsonRpcVersion,
 		ID:      uniqueRequestID,
 		Method:  vaulttypes.MethodSecretsCreate,
-		Params: &vaultcommon.CreateSecretsRequest{
-			RequestId: uniqueRequestID,
-			EncryptedSecrets: []*vaultcommon.EncryptedSecret{
-				{
-					Id: &vaultcommon.SecretIdentifier{
-						Key:   secretID,
-						Owner: owner,
-						// Namespace: "main", // Uncomment if you want to use namespaces
-					}, // Note: Namespace is not used in this test, but can be added if needed
-					EncryptedValue: encryptedSecret,
-				},
-			},
-		},
+		Params:  &secretsCreateRequestBodyJson,
 	}
-	requestBody, err := json.Marshal(secretsCreateRequest)
+	allowlistRequest(t, owner, jsonRequest, opts, wfRegistryContract)
```

Comment:
Right now, this call does request allowlisting.
It is only added for CreateSecrets in this PR, but needs to be done for all other operations too.